### ISSUE TITLE
image size

### ISSUE THREAD

**Issue by alicera** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#171]
1. In the train.py, what is the function of CropPad ?  
    If I set the crop_x = 512 and crop_y = 256, the network input size will be [3,256,512] [c,h,w], 
    right ? 

![image](https://user-images.githubusercontent.com/59388989/116839313-810ea780-ac04-11eb-8231-365f03039701.png)

2. In the val.py, the default base_height = 368
    Does it resize the height side of your images to 368, maintaining aspect ratio ?
    If I set the network input size [3,256,512] , the base_height = 256  ?
![image](https://user-images.githubusercontent.com/59388989/116839521-43f6e500-ac05-11eb-84e9-c51fce548e6e.png)

3. In the prepare_train_labels.py ,  net_input_size = 368
    what is the function of net_input_size ?  
![image](https://user-images.githubusercontent.com/59388989/116839827-4a399100-ac06-11eb-9fbc-f5fcfd5d5ed2.png)

 I am looking forward to your reply.

**Comment by Daniil-Osokin** [comment]
Hi! 1 - think so. 2 - think so. 3 - it means to which resolution scale scale the box of the current person. I any case - just follow the code.

**Comment by alicera** [comment]
Thanks for your help.

If I want to train the network input size [3,256,512].
I only adjust the function of CropPad (cropx=512, cropy=256) in train.py.
And I dont modify parameter ( net_input_size = 368) in the  prepare_train_labels.py

**Comment by Daniil-Osokin** [comment]
You can try as is for some iterations. Then prepare training labels with `net_input_size=256` and train for the same number of iterations to check which work better to run full training. Useful technique here is to visualize network input (fortunately we work with images) and check if everything is ok (or looks similar to the baseline setup).

**Comment by Daniil-Osokin** [comment]
Hope, it helped.

**Comment by soheilAppear** [comment]
> Hi! 1 - think so. 2 - think so. 3 - it means to which resolution scale scale the box of the current person. In any case - just follow the code.

I appreciate it if you answer what is the function of the CropPad? Is it going to crop the image and remove the unnecessary part of the images for a faster training process?

**Comment by Daniil-Osokin** [comment]
Network trains with batches, all items in a batch must have the same size. One way to achieve this is to do crops of the same fixed size from images. Each crop is made around person center (with some random shift for augmentation). It is better to use all persons from the training dataset, however some of them are near the image border, so such centered crop just will not fit. That is why for these cases lacking pixels (outside the image border) will be filled with pad value. You always can [check the code](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/57db93e3b7254acaa0f03bb67950738ff7d7b1a6/datasets/transformations.py#L132) for the details.
